Classes/common/touchdb/TDBlobStore.m

```diff
@@ -187,6 +190,8 @@ - (NSInteger)deleteBlobsExceptWithKeys:(NSSet*)keysToKeep withDatabase:(FMDataba
             [NSData dataWithBytes:oneRow.key.bytes length:sizeof(oneRow.key.bytes)];
         if ([keysToKeep containsObject:curKeyData]) {
             // Do not delete blob. It is an exception.
+            [filesToKeep addObject:oneRow.blobFilename];
+
             continue;
         }
 
```

Comment:
`deleteFilesExceptWithFilenames:` returns a BOOL. Also, if an attachment could not be deleted, I do not try to delete it again because it won't be deleted from db.

Let me know what you think.

Fixed in f14f8e2